src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/WebSocketsTransport.cs

```diff
@@ -113,7 +114,19 @@ public async Task StartAsync(Uri url, TransferFormat transferFormat, Cancellatio
                 var accessToken = await _accessTokenProvider();
                 if (!string.IsNullOrEmpty(accessToken))
                 {
-                    _webSocket.Options.SetRequestHeader("Authorization", $"Bearer {accessToken}");
+                    // We can't use request headers in the browser, so instead append the token as a query string if running WASM
+                    if (Utils.IsRunningBlazorWasm())
```

Comment:
We should cache this in the constructor or even as a private static on the class